wp_api/src/uniffi_serde.rs

```diff
@@ -0,0 +1,30 @@
+#[derive(Debug, uniffi::Object, thiserror::Error)]
+#[error("{inner:?}")]
+pub(crate) struct UniffiSerializationError {
+    inner: serde_json::Error,
+}
+
+impl From<serde_json::Error> for UniffiSerializationError {
+    fn from(inner: serde_json::Error) -> Self {
+        Self { inner }
+    }
+}
+
+#[macro_export]
+macro_rules! uniffi_export_serialization {
+    ($func_name:ident, $type:ty) => {
+        paste::paste! {
+            #[uniffi::export]
+            fn [<serialize_ $func_name>](value: $type) -> Result<Vec<u8>, Arc<$crate::uniffi_serde::UniffiSerializationError>> {
```

Comment:
I am hoping we'll switch to `uniffi::Enum`, so `Arc` won't be necessary anymore. However, if we don't, we should use `std::sync::Arc` here. It looks like [I didn't do this correctly in `api_client`](https://github.com/Automattic/wordpress-rs/blob/22e854728af76c230b3feae92193273fe102242d/wp_api/src/api_client.rs#L183) which I'll open a PR to fix.